app/views/comments/_comment.html.erb

```diff
@@ -1 +1,7 @@
-<p><%= comment.user.name %> => <%= comment.text %></p>
\ No newline at end of file
+<p><%= comment.user.name %> => <%= comment.text %></p>
+
+<%= button_to 'Delete Comment',
+              user_post_comment_path(user_id: comment.user_id, post_id: comment.post_id, id: comment.id),
+              method: :delete,
+              data: { confirm: 'Are you sure you want to delete this comment?' },
+              class: "delete-comment-btn" if current_user == comment.user || current_user.role == 'admin' %>
```

Comment:
- [ ] The purpose of the ability authorization file is ensure we don't need to repeat this code. All rules have been defined in that file (if any rule does not work as expected then only that file should be modified). Kindly use the `can?` method as provided by cancancan gem so we are sure the rules are being used.